receiver/hostmetricsreceiver/internal/scraper/processscraper/process.go

```diff
@@ -0,0 +1,54 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processscraper
+
+import (
+	"github.com/shirou/gopsutil/cpu"
+	"github.com/shirou/gopsutil/process"
+)
+
+type processHandle interface {
+	GetPid() int32
+	Name() (string, error)
+	Exe() (string, error)
+	Username() (string, error)
+	Cmdline() (string, error)
+	Times() (*cpu.TimesStat, error)
+	MemoryInfo() (*process.MemoryInfoStat, error)
+	IOCounters() (*process.IOCountersStat, error)
+}
+
+type gopsProcessHandle struct {
+	*process.Process
+}
+
+func (p *gopsProcessHandle) GetPid() int32 {
+	return p.Pid
+}
+
+var getProcessHandles = getProcessHandlesInternal
+
+func getProcessHandlesInternal() ([]processHandle, error) {
+	processes, err := process.Processes()
+	if err != nil {
+		return nil, err
+	}
+
+	processAccessors := make([]processHandle, len(processes))
+	for i, proc := range processes {
+		processAccessors[i] = &gopsProcessHandle{proc}
+	}
+	return processAccessors, nil
+}
```

Comment:
This function exists only to convert the slice returned by `gopsutil/process.Processes()` into a slice of the interface type declared above so that we can write tests against this. I'm not aware of a better way to do this in Go but happy to learn if there is a better alternative.

Reply:
You can make the getProcessHandlesInternal a function pointer that you replace that pointer so you can return the initial struct not interface but you can mock the results. Does that work?

Reply:
I don't think that will work because Go doesn't support covariance of slices.

```
cannot use processes (variable of type []*process.Process) as []processHandle value in return statement
```

Reply:
I've added a second commit that refactors this function to return an interface that represents a slice (with `At()` & `Len()` funcs) so that we only need to wrap the returned slice once instead of wrapping each element. This seems to be a better trade off.

The `gopsutil/process.Process` struct has a Pid property rather than an accessor function, so I had to make accessing the Pid a function on the root struct which is slightly odd but doesn't effect readability too much.

Reply:
Please comment the struct that it exists just for testing.